cmd/openshift-tuned.go

```diff
@@ -652,128 +411,103 @@ func timedTunedReloader(tuned *tunedState) (err error) {
 
 	err = profilesExtract()
 	if err != nil {
 		return err
 	}
 
-	config, err := getConfig()
+	kubeConfig, err := getConfig()
 	if err != nil {
 		return err
 	}
 
-	clientset, err := kubernetes.NewForConfig(config)
+	cs, err := tunedclientset.NewForConfig(kubeConfig)
 	if err != nil {
 		return err
 	}
 
-	// Create a ticker to do a full node/pod labels pull
-	tickerPull := time.NewTicker(time.Second * time.Duration(resyncPeriod.now))
-	defer tickerPull.Stop()
-	glog.Infof("Resync period to pull node/pod labels: %d [s]", resyncPeriod.now)
+	// Perform an initial list and start a watch on Profiles in "namespace"
+	profileLW := cache.NewListWatchFromClient(cs.TunedV1().RESTClient(), "Profiles", namespace, fieldSelector)
 
-	// When the first pod pull should happen; try to avoid lockstep
-	tuned.setNextPodLabelsPullTime(time.Now().Unix())
+	si := cache.NewSharedInformer(profileLW, &tunedv1.Profile{}, 0)
+	si.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			profile, err := getTunedProfile(obj)
+			if err != nil {
+				klog.Errorf("%s", err.Error())
+				return
+			}
+			klog.V(1).Infof("profile %q added, tuned profile requested: %s", profile.ObjectMeta.Name, profile.Spec.Config.TunedProfile)
+			disableSystemTuned() // Whe moving this call elsewhere, remember it is undesirable to disable system tuned
+			// on nodes that should not be handled by openshift-tuned
+			tunedRecommendFileWrite(profile.Spec.Config.TunedProfile)
+			tuned.change.profile = true
+		},
+		DeleteFunc: func(obj interface{}) {
+			profile, err := getTunedProfile(obj)
+			if err != nil {
+				klog.Errorf("%s", err.Error())
+				return
+			}
+			klog.V(1).Infof("profile %q deleted, keeping the old tuned profile: %s", profile.ObjectMeta.Name, profile.Spec.Config.TunedProfile)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			profile, err := getTunedProfile(newObj)
+			if err != nil {
+				klog.Errorf("%s", err.Error())
+				return
+			}
+			klog.V(1).Infof("profile %q changed, tuned profile requested: %s", profile.ObjectMeta.Name, profile.Spec.Config.TunedProfile)
+			tunedRecommendFileWrite(profile.Spec.Config.TunedProfile)
+			tuned.change.profile = true
+		},
+	})
 
-	// Pull node and pod labels before entering the loop; node labels would otherwise be fetched after resyncPeriod.now
-	if err := pullLabels(clientset, &tuned, nodeName); err != nil {
-		return err
-	}
+	stop := make(chan struct{})
+	defer close(stop)
+	go si.Run(stop)
 
-	// Create a ticker to dump node/pod labels, extract new profiles and possibly reload tuned;
-	// this also rate-limits reloads to a maximum of labelDumpInterval reloads/s
-	tickerReload := time.NewTicker(time.Second * time.Duration(labelDumpInterval))
+	// Create a ticker to extract new profiles and possibly reload tuned;
+	// this also rate-limits reloads to a maximum of profileExtractInterval reloads/s
+	tickerReload := time.NewTicker(time.Second * time.Duration(profileExtractInterval))
```

Comment:
Why did we want to shorten the ticker time to every 1s?

Reply:
Because I believe we can now.  Faster change propagation.